ui/app/components/app/modals/edit-approval-permission/edit-approval-permission.component.js

```diff
@@ -0,0 +1,161 @@
+                }
+              </div>
+              <div className="edit-approval-permission__edit-section__option-description" >
+                Spend limit requested by Uniswap
+              </div>
+              <div className="edit-approval-permission__edit-section__option-value" >
+                {`${tokenAmount} ${tokenSymbol}`}
+              </div>
+            </div>
+          </div>
+          <div className="edit-approval-permission__edit-section__option">
+            <div
+              className="edit-approval-permission__edit-section__radio-button"
+              onClick={() => this.setState({ selectedOption: 'custom' })}
+            >
+              <div className={classnames({
+                'edit-approval-permission__edit-section__radio-button-outline': this.state.selectedOption !== 'custom',
+                'edit-approval-permission__edit-section__radio-button-outline--selected': this.state.selectedOption === 'custom',
+              })} />
+              <div className="edit-approval-permission__edit-section__radio-button-fill" />
+              {this.state.selectedOption === 'custom' && <div className="edit-approval-permission__edit-section__radio-button-dot" />}
+            </div>
+            <div className="edit-approval-permission__edit-section__option-text">
+              <div className={classnames({
+                'edit-approval-permission__edit-section__option-label': this.state.selectedOption !== 'custom',
+                'edit-approval-permission__edit-section__option-label--selected': this.state.selectedOption === 'custom',
+              })}>
+                Custom spend limit
+              </div>
+              <div className="edit-approval-permission__edit-section__option-description" >
+                Enter a max spend limit
+              </div>
+              <div className="edit-approval-permission__edit-section__option-input" >
+                <TextField
+                  type="number"
+                  min="0"
+                  placeholder={ `${customTokenAmount || tokenAmount} ${tokenSymbol}` }
+                  onChange={(event) => this.setState({ customSpendLimit: event.target.value })}
+                  fullWidth
+                  margin="dense"
+                  value={ this.state.customSpendLimit }
+                />
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    )
+  }
+
+  render () {
+    const { t } = this.context
+    const { setCustomAmount, hideModal } = this.props
+
+    return (
+      <Modal
+        onSubmit={() => {
+          setCustomAmount(Number(this.state.customSpendLimit))
+          hideModal()
+        }}
+        submitText={t('save')}
+        submitType="primary"
+        contentClass="edit-approval-permission-modal-content"
+        containerClass="edit-approval-permission-modal-container"
+        submitDisabled={ this.state.customSpendLimit === this.props.customTokenAmount }
```

Comment:
This prevents saving `Unlimited` if the custom amount selected is equal to the custom token amount.

e.g.

1. Set a custom spend limit of 5, then save it
2. Edit permissions again
3. Set a custom limit of 5
4. Select 'Unlimited'

Save is then disabled, when it should be enabled.

Reply:
Actually, the `onSubmit` handler doesn't seem to allow setting 'Unlimited' anyway :thinking:

Reply:
This is indeed curious